go/process/health_dao.go

```diff
@@ -147,10 +167,18 @@ func expireAvailableNodes() error {
 			`,
 		registrationPollSeconds*2,
 	)
-	return log.Errore(err)
+	if err != nil {
+		log.Errorf("ExpireAvailableNodes: failed to remove old entries: %+v", err)
+	}
+
+	// show we're done
+	expiryMutex.Lock()
+	expiryRunning = false
+	expiryMutex.Unlock()
```

Comment:
There is no harm per se, just the complication of the code. We would need to create 30 such mutexes and duplicate this code all over the place.

For this, I don't see the need for paranoia. Again, if you hit this, then your problems are greater.

The solution you offer is slightly complicated in terms of coding. You can achieve the same with two lines of code via https://golang.org/pkg/sync/atomic/#CompareAndSwapInt64, or consolidate the `Lock/Unlock()` code, or refactor it into a class. As it is, it is _correct_, but something that I wouldn't want to see in this function. 

We can take this offline to discuss how the same can be solved in other ways, but meanwhile let's please remove it.